pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -269,42 +270,62 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundVirtualHosts(node *mod
 	virtualHostWrappers := istio_route.BuildSidecarVirtualHostsFromConfigAndRegistry(node, push, nameToServiceMap,
 		virtualServices, listenerPort)
 	vHostPortMap := make(map[int][]*route.VirtualHost)
-	uniques := make(map[string]struct{})
+
+	vhosts := sets.Set{}
+	vhdomains := sets.Set{}
+
 	for _, virtualHostWrapper := range virtualHostWrappers {
 		// If none of the routes matched by source, skip this virtual host
 		if len(virtualHostWrapper.Routes) == 0 {
 			continue
 		}
-
 		virtualHosts := make([]*route.VirtualHost, 0, len(virtualHostWrapper.VirtualServiceHosts)+len(virtualHostWrapper.Services))
+
 		for _, hostname := range virtualHostWrapper.VirtualServiceHosts {
 			name := domainName(hostname, virtualHostWrapper.Port)
-			if _, found := uniques[name]; !found {
-				uniques[name] = struct{}{}
+			duplicate := duplicateVirtualHost(name, vhosts)
+			if !duplicate {
+				domains := []string{hostname, name}
+				dl := len(domains)
+				domains = dedupeDomains(domains, vhdomains)
+				if dl != len(domains) {
+					duplicate = true
```

Comment:
is this just producing the metric below? seems like we don't exit if its a duplicate?

Reply:
If virtual host it self is duplicate we do not add it to the list of virtual hosts (current behaviour) and if domain is duplicate across virtual hosts - we skip the domain (new behaviour). In both the cases it increments the metric.